ffmuc-mesh-vpn-wireguard-vxlan/files/lib/gluon/gluon-mesh-wireguard-vxlan/checkuplink

```diff
@@ -18,24 +18,42 @@ clean_port() {
 	echo "$(echo $1 | sed -r 's/:[0-9]+$|\[|\]//g')"
 }
 
-check_address_family() {
-	local peer_publickey="$1"
-	local peer_endpoint="$2"
-	local gateway="$(clean_port $peer_endpoint)"
+extract_port() {
+    local address_port="$1"
+
+    # Remove brackets for IPv6 addresses
+    local address="$(echo "$address_port" | sed -e 's/\[//;s/\]//')"
+
+    # Extract port number
+    echo "$address" | awk -F: '{print $NF}'
+}
+
+combine_ip_port() {
+    local ip="$1"
+    local port="$2"
+
+    if echo "$ip" | grep -q ":"; then
+        echo "[$ip]:$port"
+    else
+        echo "$ip:$port"
+    fi
+}
+
+resolve_host() {
+	local peer_host="$1"
 	# Check if we have a default route for v6 if not fallback to v4
 	defgw=$(ip -6 route show table 1 | grep 'default via')
 	if [ "$?" -eq "0" ]; then
-		local ipv6="$(gluon-wan nslookup $gateway | grep 'Address:\? [0-9]' | egrep -o '([a-f0-9:]+:+)+[a-f0-9]+')"
-		echo [$ipv6]$(echo $peer_endpoint | egrep -oe :[0-9]+$)
+		local ipv6="$(gluon-wan nslookup $peer_host | grep 'Address:\? [0-9]' | egrep -o '([a-f0-9:]+:+)+[a-f0-9]+')"
+		echo $ipv6
 	else
-		local ipv4="$(gluon-wan nslookup $gateway | grep 'Address:\? [0-9]' | grep -oE "\b([0-9]{1,3}\.){3}[0-9]{1,3}\b")"
-		echo $ipv4$(echo $peer_endpoint | egrep -oe :[0-9]+$)
+		local ipv4="$(gluon-wan nslookup $peer_host | grep 'Address:\? [0-9]' | grep -oE "\b([0-9]{1,3}\.){3}[0-9]{1,3}\b")"
+		echo $ipv4
 	fi
-
 }
 
 # Do we already have a private-key? If not generate one
-temp=$(uci get wireguard.mesh_vpn.privatekey)
+uci get wireguard.mesh_vpn.privatekey >/dev/null
```

Comment:
This should not be needed with the latest version of checkuplink (November 2023).